packages/lucide-react/build.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { nodeFs } from '@file-services/node';
+import * as lucideReact from 'lucide-react';
+
+interface IconData {
+    name: string;
+    aliases: string[];
+    svgString: string;
+}
+
+interface IconListEntry {
+    name: string;
+    aliases: string[];
+    file: string;
+}
+
+interface IconTypeAugmentation {
+    name: string;
+    aliases: string[];
+    file: string;
+}
+
+interface TypeAugmentationFile {
+    icons: IconTypeAugmentation[];
+}
+
+function buildIcons(): void {
+    const distDir = nodeFs.join(__dirname, 'dist');
+    const svgDir = nodeFs.join(distDir, 'svg');
+
+    nodeFs.ensureDirectorySync(distDir);
+    nodeFs.ensureDirectorySync(svgDir);
+
+    const svgsByContent = new Map<string, IconData>();
+
+    for (const [exportName, IconComponent] of Object.entries(lucideReact)) {
+        // Skip non-component exports
+        if (typeof IconComponent !== 'function' && typeof IconComponent !== 'object') {
+            continue;
+        }
+
+        if (!IconComponent) {
+            continue;
+        }
+
+        // Skip known utility functions
+        if (exportName === 'createLucideIcon' || exportName === 'Icon') {
+            continue;
+        }
+
+        // Icon components have uppercase first letter
+        if (exportName[0] !== exportName[0].toUpperCase()) {
+            continue;
+        }
+
+        try {
+            const svgString = renderToStaticMarkup(
+                React.createElement(IconComponent as React.ComponentType<{ size: number; strokeWidth: number }>, {
+                    size: 24,
+                    strokeWidth: 2,
+                }),
+            );
+
+            // Check if we've seen this SVG content before
+            if (svgsByContent.has(svgString)) {
+                const existing = svgsByContent.get(svgString)!;
+                // Keep the shortest name
+                if (exportName.length < existing.name.length) {
```

Comment:
while this works for the Abc/LucideAbc/AbcIcon case, this makes no sense for icons that are being renamed (with the old one kept around)

Reply:
iterate on `lucideReact.icons` and avoid checking for same-content and aliases.